scripts/meta_analysis.py

```diff
@@ -89,11 +113,18 @@ def inv_var_meta( studies : List[Tuple['Study','VariantData']], is_het_test = Fa
         het_p=het_test(effs_size_org, weights, beta_meta)
     else:
         het_p=None
-    return (beta_meta, math.sqrt(1/sum_inv_var), max(sys.float_info.min * sys.float_info.epsilon, 2 * scipy.stats.norm.sf(abs(sum(effs_inv_var) / math.sqrt(sum_inv_var) ))), het_p) if len(effs_inv_var)==len(studies) else None
-
+    return (beta_meta, math.sqrt(1/sum_inv_var), max(sys.float_info.min * sys.float_info.epsilon, 2 * scipy.stats.norm.sf(abs(sum(effs_inv_var) / math.sqrt(sum_inv_var) ))), het_p) if len(effs_inv_var)==len(studies) else (None, None, None, None)
```

Comment:
the reason I don't like repeating theNone values to form a tuple is that a single None indicates that the data could not be computed at all. Returning None or possibly raising exceptions. In this way the caller can act accordingly.... it just needs to be documented that in case value can't be computed a None is returned.
Off course you could argue that you can check None values form the tuple... which of the values should you check? can some of them be None and some not? It is much more clear to return just None, so I would revert back to your previous implementation of checking if the value was none and then doing something on top of that instead of printing whatever the method returns in case when it can't compute the value.

Reply:
Good point, I totally agree! I'll revert back to the original.